chia/full_node/mempool.py

```diff
@@ -264,6 +273,44 @@ def add_to_pool(self, item: MempoolItem) -> None:
         assert item.cost <= self.mempool_info.max_block_clvm_cost
 
         with self._db_conn:
+            # we have certain limits on transactions that will expire soon
+            # (in the next 15 minutes)
+            block_cutoff = self._block_height + 48
+            time_cutoff = self._timestamp + 900
+            if (item.assert_before_height is not None and item.assert_before_height < block_cutoff) or (
+                item.assert_before_seconds is not None and item.assert_before_seconds < time_cutoff
+            ):
+                # this lists only transactions that expire soon, in order of
+                # lowest fee rate along with the cumulative cost of such
+                # transactions counting from highest to lowest fee rate
+                cursor = self._db_conn.execute(
+                    """
+                    SELECT name,
+                        fee_per_cost,
+                        SUM(cost) OVER (ORDER BY fee_per_cost DESC, seq ASC) AS cumulative_cost
+                    FROM tx
+                    WHERE assert_before_seconds IS NOT NULL AND assert_before_seconds < ?
+                        OR assert_before_height IS NOT NULL AND assert_before_height < ?
+                    ORDER BY cumulative_cost DESC
+                    """,
```

Comment:
@trepca please review this query with some care. As far as I can tell (from the tests) the `SUM(cost) OVER ...` field is still constrained by the outer `WHERE` clause, is that right?

Reply:
lgtm